src/models/products.model.js

```diff
@@ -0,0 +1,18 @@
+const { AppDataSource } = require("./data-source");
+
+async function getProductById(productId) {
+  const products = await AppDataSource.query(`
+    SELECT
+      id
+    FROM
+      products
+    WHERE
+      id = "${productId}"
+    ;
+  `);
+  return products[0];
+}
+
```

Comment:
```suggestion
const getProductById = async (productId) => {
  const products = await AppDataSource.query(`
    SELECT
      id
    FROM
      products
    WHERE
      id = "${productId}"
    ;
  `);
  return products[0];
}

```

ES6 이후 부터는 화살표 함수로 작성하시는 것이 좋습니다. 최신 문법!
function 문법은 hoisting 문제가 발생합니다 :) hoisting에 대해서 한 번 학습해 보시는 것을 추천드립니다.

Reply:
function 문법이 오히려 호이스팅 문제가 없는 것으로 알고 있습니다.
https://developer.mozilla.org/ko/docs/Web/JavaScript/Guide/Functions#function_hoisting
오히려 const로 화살표함수를 사용하면 선언은 되지만 값이 할당되지는 않는 문제가 생길 것 같습니다.
https://developer.mozilla.org/ko/docs/Web/JavaScript/Guide/Grammar_and_types#variable_hoisting

<img width="851" alt="image" src="https://github.com/wecode-bootcamp-korea/49-2nd-5Sherlock-backend/assets/52498558/6b8fc0c9-7c8b-4410-b991-972e2ac36f08">